example/glue/objectdescription.h

```diff
@@ -19,17 +19,26 @@
  * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
  *******************************************************************************/
 
-#if !defined(OBJECTDESCRIPTION_H_)
-#define OBJECTDESCRIPTION_H_
+#if !defined(OMRAPP_OBJECTDESCRIPTION_H_)
+#define OMRAPP_OBJECTDESCRIPTION_H_
+
+#include "omrcfg.h"
 
 #include "omrcomp.h"
 #include "omr.h"
 
 /**
  * Object token definitions to be used by OMR components.
  */
-typedef uintptr_t* languageobjectptr_t;
-typedef uintptr_t* omrobjectptr_t;
+
+namespace OMRApp {
+
+class Object;
+
+}  /* namespace OMRApp */
+
+typedef OMRApp::Object* languageobjectptr_t;
```

Comment:
Will this typedef work if the file is included in a C file and not a C++ file?  I believe you have properly defined it but I am just wondering if you have verified that it will work in a C file.

Reply:
No this won't work--namespaces will break it. Is c compatibility needed/wanted in the example? I can change this, but it's nice that the `objectptr` types are literally `OMRApp::Object *`.